src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -375,7 +375,8 @@ private static object BindToCollection(Type type, IConfiguration config, BinderO
                         return; // We are already done if binding to a new collection instance worked
                     }
 
-                    bindingPoint.SetValue(CreateInstance(type));
+                    var instance = CreateInstance(type, config, options);
```

Comment:
I think this use of `var` violates the repo style guidelines; probably we could just inline the call to CreateInstance as it was previously.